### PR TITLE
Ankit | Stage | Prod - Confirmation pop up should come on delete session option

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cw54agw

### CODE REVIEW

**Review by dvCodeWorld — APPROVED** [review]

File: apps/web-giddh/src/app/subscription/subscription.component.ts [hunk 1]
```diff
@@ -311,11 +315,56 @@ export class SubscriptionComponent implements OnInit, OnDestroy {
             sessionId,
             sessionIndex
         };
-        this.store.dispatch(this.sessionAction.deleteSession(requestPayload));
+        const dialogRef = this.deleteConfirmationDialog(false);
+        dialogRef.afterClosed().pipe().subscribe(response => {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/subscription/subscription.component.ts]:
```suggestion
        dialogRef.afterClosed().pipe(take(1)).subscribe(response => {
```

File: apps/web-giddh/src/app/subscription/subscription.component.ts [hunk 2]
```diff
@@ -311,11 +315,56 @@ export class SubscriptionComponent implements OnInit, OnDestroy {
...
+    }
+
+    /**
+     * Deletes All sessions
+     *
+     * @memberof SubscriptionComponent
+     */
+    public clearAllSession(): void {
+        const dialogRef = this.deleteConfirmationDialog(true);
+        dialogRef.afterClosed().pipe().subscribe(response => {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/subscription/subscription.component.ts]:
```suggestion
        dialogRef.afterClosed().pipe(take(1)).subscribe(response => {
```

File: apps/web-giddh/src/app/subscription/subscription.component.ts [hunk 3]
```diff
@@ -311,11 +315,56 @@ export class SubscriptionComponent implements OnInit, OnDestroy {
...
+            if (response === this.commonLocaleData?.app_yes) {
+                this.store.dispatch(this.sessionAction.deleteAllSession());
+            }
+        });
     }
 
-    public clearAllSession() {
-        this.store.dispatch(this.sessionAction.deleteAllSession());
+    /**
+     *
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/subscription/subscription.component.ts]: Documentation

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/subscription/subscription.component.ts [hunk 4]
```diff
@@ -311,11 +315,56 @@ export class SubscriptionComponent implements OnInit, OnDestroy {
...
+     *
+     * @param {boolean} allDeletes
+     * @return {*}  {*}
+     * @memberof SubscriptionComponent
+     */
+    public deleteConfirmationDialog(allDeletes: boolean): any {
+        const buttons: Array<ConfirmationModalButton> = [{
+            text: this.commonLocaleData?.app_yes,
+            color: 'primary'
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/subscription/subscription.component.ts]: move to general service

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/subscription/subscription.component.ts (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/subscription/subscription.component.ts]: Done

File: apps/web-giddh/src/app/subscription/subscription.component.ts (hunk 3 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/subscription/subscription.component.ts]: Done

File: apps/web-giddh/src/app/subscription/subscription.component.ts (hunk 1 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/subscription/subscription.component.ts]: Done

File: apps/web-giddh/src/app/subscription/subscription.component.ts (hunk 4 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/subscription/subscription.component.ts]: Done

**Pull request closed without merging**